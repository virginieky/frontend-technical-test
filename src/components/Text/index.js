import styled from 'styled-components';
import PropTypes from 'prop-types';

const Text = styled.p`
  line-height: ${({ lineHeight }) => lineHeight};
  color: ${({ color }) => color};
  font-size: ${({ fontSize }) => fontSize};
  font-weight: ${({ fontWeight }) => fontWeight};
  text-transform: ${({ textTransform }) => textTransform};
  ${({ ellipsis }) => ellipsis && `
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  `}
  margin-bottom: 0;
`;

Text.defaultProps = {
  color: 'black',
  ellipsis: false,
  fontSize: '1.3rem',
  fontWeight: 'regular',
  lineHeight: '1.3rem',
  textTransform: 'none',
};

Text.propTypes = {
  color: PropTypes.string,
  ellipsis: PropTypes.bool,
  fontSize: PropTypes.string,
  fontWeight: PropTypes.string,
  lineHeight: PropTypes.string,
  textTransform: PropTypes.string,
};

export default Text;
