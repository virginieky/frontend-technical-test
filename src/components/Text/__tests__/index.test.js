import React from 'react';
import { render } from '@testing-library/react';
import Text from '../index';

describe('components | Text', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Text />);

    expect(asFragment()).toMatchSnapshot();
  });
});
