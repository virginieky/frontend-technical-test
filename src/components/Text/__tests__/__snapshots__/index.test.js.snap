// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Text should match the snapshot 1`] = `
<DocumentFragment>
  <p
    class="Text-sc-yw3xjc-0 ihAFkw"
    color="black"
    font-size="1.3rem"
    font-weight="regular"
  />
</DocumentFragment>
`;
